src/types/mod.rs

```diff
@@ -8,9 +8,18 @@ mod script;
 mod serde;
 mod transaction;
 
-#[derive(Deserialize, Serialize, Debug, PartialEq)]
+use bitcoin::util::hash::{Sha256dHash, HexError};
+
+#[derive(Clone, Deserialize, Serialize, Debug, PartialEq)]
 pub struct BlockHash(String);
```

Comment:
Since recently, `rust-bitcoin` already ships with an implementation of `Serialize` and `Deserialize` for `Sha256dHash`.

The `BlockHash` struct was originally only created because `Sha256dHash` was not directly serializable. Could you try, if we can actually get rid of the `BlockHash` struct altogether and simply replace it with `Sha256dHash`?

If not, we could at least replace the inner representation of `Blockhash` with a `Sha256dHash` to catch the error of invalid hex at deserialization time.